packages/xrpl/test/models/typeUtility.test.ts

```diff
@@ -0,0 +1,191 @@
+    assert.typeOf(omitted.id, 'number')
+
+    // Verify api_version properties remain
+    assert.property(omitted, 'api_version')
+    assert.typeOf(omitted.api_version, 'number')
+
+    type HasCommand = 'command' extends keyof BaseRequest ? true : false
+    type HasNoCommand = 'command' extends keyof RequestMinusCommand
+      ? false
+      : true
+
+    assertType<true>({} as HasCommand)
+    assertType<true>({} as HasNoCommand)
+  })
+
+  it(`partial utility`, function () {
+    type OptionalRequest = Partial<BaseRequest>
+
+    // Verify Partial makes all properties optional
+    const empty: OptionalRequest = {}
+    assert.notProperty(empty, 'command')
+    assert.notProperty(empty, 'id')
+    assert.notProperty(empty, 'api_version')
+
+    // Verify optional nature at compile-time
+    // This should compile without error
+    const minimal: OptionalRequest = {}
+    assert.isUndefined(minimal.command)
+  })
+})
+
+describe('AccountInfoRequest', function () {
+  it(`pick utility`, function () {
+    // Define a value with the picked type
+    const picked: Pick<AccountInfoRequest, 'command'> = {
+      command: 'account_info',
+    }
+
+    // Verify the property exists
+    assert.property(picked, 'command')
+    assert.typeOf(picked.command, 'string')
+
+    // Verify TypeScript enforces the picked type
+    assert.isUndefined((picked as any).id)
+  })
+
+  it(`omit utility`, function () {
+    type RequestMinusCommand = Omit<AccountInfoRequest, 'command'>
+
+    const control: AccountInfoRequest = {
+      command: 'account_info',
+      account: 'account_address',
+    }
+    // Define a value with the omitted type
+    const omitted: RequestMinusCommand = {
+      account: 'account_address',
+      id: 1,
+      api_version: 2,
+    }
+
+    // Verify command is not present
+    assert.notProperty(omitted, 'command')
+    assert(typeof control.command === 'string')
+
+    // Verify id properties remain
+    assert.property(omitted, 'id')
+    assert.typeOf(omitted.id, 'number')
+
+    // Verify api_version properties remain
+    assert.property(omitted, 'api_version')
+    assert.typeOf(omitted.api_version, 'number')
+
+    type HasCommand = 'command' extends keyof AccountInfoRequest ? true : false
+    type HasNoCommand = 'command' extends keyof RequestMinusCommand
+      ? false
+      : true
+
+    assertType<true>({} as HasCommand)
+    assertType<true>({} as HasNoCommand)
+  })
+
+  it(`partial utility`, function () {
+    type OptionalRequest = Partial<AccountInfoRequest>
+
+    // Verify Partial makes all properties optional
+    const empty: OptionalRequest = {}
+    assert.notProperty(empty, 'command')
+    assert.notProperty(empty, 'id')
+    assert.notProperty(empty, 'api_version')
+
+    // Verify optional nature at compile-time
+    // This should compile without error
+    const minimal: OptionalRequest = {}
+    assert.isUndefined(minimal.command)
+  })
+})
+
+describe('Stream', function () {
```

Comment:
Since there is no update to the `Stream` type, I don't think these tests are needed. Am I missing something?

I understand that `Stream` inherits from `BaseRequest`, however there are many classes which inherit from `BaseRequest`.

Reply:
Okay I am just going to remove it the testing for everything, except the BaseRequest.